api4/user_test.go

```diff
@@ -2451,6 +2451,39 @@ func TestRevokeAllSessions(t *testing.T) {
 	CheckUnauthorizedStatus(t, resp)
 }
 
+func TestRevokeSessionsFromAllUsers(t *testing.T) {
+	th := Setup().InitBasic()
+	defer th.TearDown()
+
+	user := th.BasicUser
+	th.Client.Login(user.Email, user.Password)
+	_, resp := th.Client.RevokeSessionsFromAllUsers()
+	CheckForbiddenStatus(t, resp)
+
+	th.Client.Logout()
+	_, resp = th.Client.RevokeSessionsFromAllUsers()
+	CheckUnauthorizedStatus(t, resp)
+
+	admin := th.SystemAdminUser
+	th.Client.Login(admin.Email, admin.Password)
+	_, resp = th.Client.RevokeSessionsFromAllUsers()
+	CheckNoError(t, resp)
+
+	_, resp = th.Client.RevokeSessionsFromAllUsers()
+	CheckUnauthorizedStatus(t, resp)
+
+	sessions, _ := th.SystemAdminClient.GetSessions(user.Id, "")
```

Comment:
I would have expected this API call to fail with an error because all the sessions were revoked. Am I missing something?

Reply:
Good catch, but I'm afraid I have no answer for this, would the admin client do a login again? butn then it should fail as it would contain 1 session. Is this stubbed? I don't think so, but then I'm not sure the test would be useful. As I said I've looked at various options but I'm not really sure whats going on.
What I realised with your spot-on question, is that I wasn't testing this properly, so I'm changing how to check this.